kernel/packages/shared/profiles/ProfileAsPromise.ts

```diff
@@ -25,3 +25,20 @@ export function ProfileAsPromise(userId: string, version?: number): Promise<Prof
     store.dispatch(profileRequest(userId))
   })
 }
+
+export function EnsureProfile(userId: string, version?: number): Promise<Profile> {
+  const store: Store<RootState> = globalThis.globalStore
+  const existingProfile = getProfile(store.getState(), userId)
+  if (existingProfile && (!version || existingProfile.version >= version)) {
```

Comment:
Can we use a temporary bool for the `version || existingProfile.version >= version` part? Or maybe a comment?

That embedded conditional makes the logic a bit opaque to me.

Reply:
You're right. I have extracted this conditional into a variable and I think the code is clearer now. Thanks Brian!